### PR TITLE
Refactor creating otel collector configuration

### PR DESCRIPTION
**Pull request by chrkl**
**Description**

Refactor the code that generates the otel collector configuration in the telemetry-operator to improve test coverage.

Changes proposed in this pull request:

- Decouple the creation of the collector configuration from Kubernetes objects
- Introduce types with yaml marshalling for collector configuration
- Add missing tests for collector processor configuration

**Related issue(s)**

### CODE REVIEW

File: components/telemetry-operator/controller/tracepipeline/config_builder.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+	"fmt"
+
+	"github.com/kyma-project/kyma/components/telemetry-operator/apis/telemetry/v1alpha1"
+)
+
+type TlsConfig struct {
+	Insecure bool `yaml:"insecure"`
+}
+
+type OtlpExporterConfig struct {
```
Comment (skhalash) [inline components/telemetry-operator/controller/tracepipeline/config_builder.go:13]: As discussed, let's use all caps for acronyms: `OTLP` and `TLS`

File: components/telemetry-operator/controller/tracepipeline/config_builder.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+	Exporters  ExporterConfig    `yaml:"exporters"`
+	Processors ProcessorsConfig  `yaml:"processors"`
+	Extensions ExtensionsConfig  `yaml:"extensions"`
+	Service    OtlpServiceConfig `yaml:"service"`
+}
+
+func makeReceiverConfig() ReceiverConfig {
+	return ReceiverConfig{
+		OpenCensus: map[string]any{},
+		Otlp: map[string]any{
```
Comment (skhalash) [inline components/telemetry-operator/controller/tracepipeline/config_builder.go:69]: Would it make sense to have a type, for instance, for receivers, exporters, etc? Is there any benefit in having it untyped (as a map[string]any)?

**Review by skhalash — CHANGES_REQUESTED** [review]

File: components/telemetry-operator/controller/tracepipeline/config_builder.go (hunk 2 above)
Comment (skhalash) [reply components/telemetry-operator/controller/tracepipeline/config_builder.go:69]: Ok, let's leave it like this for now.

File: components/telemetry-operator/controller/tracepipeline/config_builder.go (hunk 1 above)
Comment (chrkl) [reply components/telemetry-operator/controller/tracepipeline/config_builder.go]: Done

File: components/telemetry-operator/controller/tracepipeline/config_builder.go (hunk 2 above)
Comment (chrkl) [reply components/telemetry-operator/controller/tracepipeline/config_builder.go]: Introduced types, except for the expected empty maps.

**Review by skhalash — APPROVED** [review]

**Pull request merged**